parseDataset.js

```diff
@@ -99,10 +107,15 @@ lines.slice(TITLE_LINE + 1).forEach((line) => {
     return;
   }
   if (isNewQuantity) {
+    // Concatenate ingredient_id and quantity_unit_id to get the unique quantity_id
+    const quantity_id = currentState.ingredient_id + '.' + lineObject.quantity_unit_id;
+    lineObject.quantity_id = quantity_id;
     data.ingredients[currentState.ingredient_id].quantities.push(
-      lineObject.quantity_id
+      quantity_id
     );
-    data.quantities[lineObject.quantity_id] = {
+    // If quantity_ingredient_name is not specified, use ingredient_name
+    lineObject.quantity_ingredient_name = lineObject.quantity_ingredient_name ? lineObject.quantity_ingredient_name : currentState.ingredient_name;
```

Comment:
JS tips and tricks, if the first element is false (or empty) the OR operator return the second one

```suggestion
    lineObject.quantity_ingredient_name = lineObject.quantity_ingredient_name || currentState.ingredient_name;
```

Reply:
Perfect, thanks!